### PR TITLE
cleanup couler codegen

### LINKED ISSUE #1747: Package codegen/couler does Argo related work
By the name, the package `codgen/couler` should generate the Couler program.  However, it generates the Argo YAML file:

https://github.com/sql-machine-learning/sqlflow/blob/5ab490f740decbb8c3f293461c7920d5ef2f8979/pkg/sql/codegen/couler/codegen.go#L164

This doesn't make sense -- if we want a Argo codegen, it should be named `codegen/argo` instead of `codegen/couler`.

In particular, as we have a package named [`argo`](https://github.com/sql-machine-learning/sqlflow/blob/develop/pkg/argo/), which, by the name, should be the one doing Argo related work.

### PR DESCRIPTION
**Pull request by Yancey0623**
fixed #1747

### CODE REVIEW

**Review by wangkuiyi — DISMISSED** [review]
LGTM++

File: python/couler/couler/pyfunc.py [hunk 1]
```diff
@@ -71,9 +71,12 @@ def body(func_obj):
...
-    """Return the Python file that defines the workflow.
+def workflow_name():
+    """Return the workflow name that defines the workflow.
     """
+    wf_name = os.getenv("workflow_name")
+    if wf_name != "":
+        return wf_name
     stacks = inspect.stack()
     frame = inspect.stack()[len(stacks) - 1]
     full_path = frame[0].f_code.co_filename
```
Comment (typhoonzero) [inline python/couler/couler/pyfunc.py:82]: This function still returns filename if env `workflow_name` does not exist, so the function name `workflow_name` may not telling this detail?

File: python/couler/couler/client.py [hunk 2]
```diff
@@ -38,6 +38,8 @@ def run(args):
         cluster_config_path = args.cluster_config
         os.environ["couler_cluster_config"] = cluster_config_path
 
+    os.environ["workflow_name"] = args.workflow_name
```
Comment (typhoonzero) [inline python/couler/couler/client.py:41]: Env variables should be all capitalized, or try not use env to pass parameter?

**Review by typhoonzero — APPROVED** [review]

File: python/couler/couler/client.py (hunk 2 above)
Comment (Yancey0623) [reply python/couler/couler/client.py:41]: The same action with `cluster_config`, can refactor here in the future.

**Pull request merged**